File: Loenn/entities/Penumbra_Phantasm_Intro.lua
local utils = require("utils")
local npcSessionFlags = require("ui.forms.fields.npc_session_flags")

local penumbraPhantasmIntro = {}

penumbraPhantasmIntro.name = "MaggyHelper/Penumbra_Phantasm_Intro"
penumbraPhantasmIntro.depth = 0
penumbraPhantasmIntro.justification = {0.5, 1.0}
penumbraPhantasmIntro.texture = "objects/Ingeste/sampleEntity/idle00"

penumbraPhantasmIntro.placements = {
    {
        name = "default",
        data = {
            activationMode = "touch",
            requireFlag = "",
            completionFlag = "ch20_penumbra_phantasm_intro",
            removeAfterTrigger = true,
            showSprite = false,
            texturePath = "objects/Ingeste/sampleEntity/idle00"
        }
    }
}

penumbraPhantasmIntro.fieldInformation = {
    activationMode = {
        fieldType = "string",
        editable = false,
        options = {
            "interact",
            "touch",
            "roomEnter"
        }
    },
    requireFlag = npcSessionFlags,
    completionFlag = npcSessionFlags,
    removeAfterTrigger = {
        fieldType = "boolean"
    },
    showSprite = {
        fieldType = "boolean"
    },
    texturePath = {
        fieldType = "string"
    }
}

penumbraPhantasmIntro.fieldOrder = {
    "x",
    "y",
    "activationMode",
    "requireFlag",
    "completionFlag",
    "removeAfterTrigger",
    "showSprite",
    "texturePath"
}

function penumbraPhantasmIntro.texture(room, entity)
    return entity.texturePath or "objects/Ingeste/sampleEntity/idle00"
end

function penumbraPhantasmIntro.selection(room, entity)
    return utils.rectangle(entity.x - 8, entity.y - 16, 16, 16)
end

return penumbraPhantasmIntro

File: framework/extracted/Ahorn/entities/Penumbra_Phantasm_Intro.jl
module Penumbra_Phantasm_Intro

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/Penumbra_Phantasm_Intro" Penumbra_Phantasm_Intro(x::Integer, y::Integer, activationMode::String="touch", completionFlag::String="ch20_penumbra_phantasm_intro", removeAfterTrigger::Bool=true, requireFlag::String="", showSprite::Bool=false, texturePath::String="objects/Ingeste/sampleEntity/idle00")

const placements = Ahorn.PlacementDict(
    "default" => Ahorn.EntityPlacement(Penumbra_Phantasm_Intro)
)

function Ahorn.selection(entity::Penumbra_Phantasm_Intro)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::Penumbra_Phantasm_Intro, room::Maple.Room)
    Ahorn.drawSprite(ctx, "objects/Ingeste/sampleEntity/idle00", entity.x, entity.y)
end

end
